runtime/parachains/src/inclusion/mod.rs

```diff
@@ -575,20 +564,18 @@ impl<T: Config> Pallet<T> {
 					},
 				};
 
-				if enact_candidate {
-					let receipt = CommittedCandidateReceipt {
-						descriptor: pending_availability.descriptor,
-						commitments,
-					};
-					let _weight = Self::enact_candidate(
-						pending_availability.relay_parent_number,
-						receipt,
-						pending_availability.backers,
-						pending_availability.availability_votes,
-						pending_availability.core,
-						pending_availability.backing_group,
-					);
-				}
+				let receipt = CommittedCandidateReceipt {
+					descriptor: pending_availability.descriptor,
+					commitments,
+				};
+				let _weight = Self::enact_candidate(
```

Comment:
It is indirectly, via benchmarking of the enter extrinisic. We are returning the consumed weight [here](https://github.com/paritytech/polkadot/blob/2d26ab3e5b5051a1eefefaa24e95be9646d2df5a/runtime/parachains/src/paras_inherent/mod.rs#L558).

The manual tracking could actually be removed. Although benchmarks are really not great so we might end up doing manual tracking again at some point.

Problems with benchmarking:

1. The benchmarks themselves seem to generate non sensible data, especially with lots of parameters.
2. We are definitely massively overestimating right now, because we are taken into account the full overhead of the enter call for each individual candidate, while in reality we only consume it once for all of them.

Longer term we can squeeze out more performance by being smarter here.